strings/base_error.h

```diff
@@ -344,100 +356,104 @@ WINRT_EXPORT namespace winrt
 
     struct hresult_access_denied : hresult_error
     {
-        hresult_access_denied() noexcept : hresult_error(impl::error_access_denied) {}
-        hresult_access_denied(param::hstring const& message) noexcept : hresult_error(impl::error_access_denied, message) {}
-        hresult_access_denied(take_ownership_from_abi_t) noexcept : hresult_error(impl::error_access_denied, take_ownership_from_abi) {}
+        hresult_access_denied(WINRT_IMPL_SOURCE_LOCATION_ARGS_SINGLE_PARAM) noexcept : hresult_error(impl::error_access_denied WINRT_IMPL_SOURCE_LOCATION_FORWARD) {}
+        hresult_access_denied(param::hstring const& message WINRT_IMPL_SOURCE_LOCATION_ARGS) noexcept : hresult_error(impl::error_access_denied, message WINRT_IMPL_SOURCE_LOCATION_FORWARD) {}
+        hresult_access_denied(take_ownership_from_abi_t WINRT_IMPL_SOURCE_LOCATION_ARGS) noexcept : hresult_error(impl::error_access_denied, take_ownership_from_abi WINRT_IMPL_SOURCE_LOCATION_FORWARD) {}
     };
```

Comment:
Not required for this PR, but ideally we can just have a macro to stamp out these derived exception types. These were all hand-rolled because originally C++/WinRT was 100% free of macros. Unfortunately, that wasn't very practical and it now relies on a fair number of macros. 😢

Reply:
It is very much debatable, but I think this is still more readable than a macro would be (at the cost of increased duplication).  The new source_location macro definitely makes it uglier, but my personal opinion is that it is not yet worth such a conversion.  It is borderline, though.